### DISCUSSION TITLE
Cleanup target dir

### DISCUSSION THREAD

**Discussion by shimunn** [corrosion-rs/corrosion#539] [Q&A]
Hi,

I'm using corrosion to compile my crate as part of an larger CMake project which has been working great so far. Except for the explosion in disk usage due to CMake/corrosion creating an new `target` dir for each invokation. Is it possibe to delete the target dir once all binaries/libs have been copied?

**Comment by jschwe** [comment]
> Except for the explosion in disk usage due to CMake/corrosion creating an new target dir for each invokation

Could you clarify what you mean with that? 

Deleting the target dir would also mean that incremental builds are impossible

**Reply by shimunn** [reply]
Well I guess the parent `CMakeLists.txt` is at fault then by creating an new build directory for each invocation. I'd like to delete the target dir nonetheless, is there an variable containing the path?

**Reply by jschwe** [reply]
Yeah, if you create a new build directory - you will get a new build directory. 

Corrosion does not provide such a variable, since this is considered an implementation detail and may change.
However if you really want to do something like that, you can [look here](https://github.com/corrosion-rs/corrosion/blob/master/cmake/Corrosion.cmake#L684) to see how corrosion sets the build directory.
